frontend/app/src/helpers/helpers.ts

```diff
@@ -133,3 +132,35 @@ export const formatRelayPerson = (person: any): any => {
     route_hint: person.route_hint
   };
 };
+
+export type Roles = "ADD BOUNTY" | 
+                    "UPDATE BOUNTY" |
+                    "DELETE BOUNTY" |
+                    "PAY BOUNTY" |
+                    "ADD USER"  |
+                    "UPDATE USER" |
+                    "DELETE USER" |
+                    "ADD ROLES" |
+                    "ADD BUDGET" |
+                    "WITHDRAW BUDGET" |
+                    "VIEW REPORT";
+
+export const userHasRole = (bountyRoles : any[], userRoles: any[], role: Roles): boolean => {
+  let hasRole = false;
+  const bountyRolesMap = {};
+  const userRolesMap = {};
+
+  bountyRoles.forEach((role: any) => { 
+    bountyRolesMap[role.name] = role.name;
+  });
+
+  userRoles.forEach((user: any) => { 
+    userRolesMap[user.role] = user.role;
+  });
+
+  if(bountyRolesMap.hasOwnProperty(role) &&  userRolesMap.hasOwnProperty(role))  {
+    hasRole = true;
+  }
+  
+  return hasRole;
+}
```

Comment:
maybe I'm a little confused on this but why does the user need both the user role and bounty role?

Reply:
The user needs the user role, and bounty role as an extra check to be sure all roles on the user role exists on the bounty roles.